launch_frontend/launch_frontend/parser.py

```diff
@@ -0,0 +1,150 @@
+    return action_parse_methods[entity.type_name](entity)
+
+
+def parse_substitution(value: Text, frontend: Text) -> launch.SomeSubstitutionsType:
+    """Parse a substitution, using its registered parsing method."""
+    if not extensions_loaded:
+        load_parser_extensions()
+    if frontend in interpolation_fuctions:
+        return interpolation_fuctions[frontend](value)
+    else:
+        return default_substitution_interpolation(value)
+
+
+def parse_description(entity: Entity) -> launch.LaunchDescription:
+    """Parse a launch description."""
+    if entity.type_name != 'launch':
+        raise RuntimeError('Expected \'launch\' as root tag')
+    actions = [parse_action(child) for child in entity.children]
+    return launch.LaunchDescription(actions)
+
+
+def str_to_bool(string):
+    """Convert xs::boolean to python bool."""
+    if not string or string.lower() in ('0', 'false'):
+        return False
+    if string.lower() in ('1', 'true'):
+        return True
+    raise RuntimeError('Expected "true" or "false", got {}'.format(string))
+
+
+def load_optional_attribute(
+    kwargs: dict,
+    entity: Entity,
+    name: Text,
+    *,
+    subst: bool = True,
+    constructor_name: Optional[Text] = None
+):
+    """Load an optional attribute of `entity` named `name` in `kwargs`."""
+    attr = getattr(entity, name, None)
+    key = name
+    if constructor_name is not None:
+        key = constructor_name
+    if attr is not None:
+        if subst:
+            kwargs[key] = parse_substitution(attr, entity.frontend)
+        else:
+            kwargs[key] = attr
+
+
+@expose_action('executable')
+def parse_executable(entity: Entity):
+    """Parse executable tag."""
+    cmd = parse_substitution(entity.cmd, entity.frontend)
+    kwargs = {}
+    load_optional_attribute(kwargs, entity, 'cwd')
+    load_optional_attribute(kwargs, entity, 'name')
+    load_optional_attribute(
+        kwargs,
+        entity,
+        'launch-prefix',
+        constructor_name='prefix'
+    )
+    load_optional_attribute(
+        kwargs,
+        entity,
+        'output',
+        subst=False)
+    shell = getattr(entity, 'shell', None)
+    if shell is not None:
+        kwargs['shell'] = str_to_bool(shell)
+    # TODO(ivanpauno): How will predicates be handle in env?
+    # Substitutions aren't allowing conditions now.
+    env = getattr(entity, 'env', None)
+    if env is not None:
+        env = {e.name: parse_substitution(e.value, e.frontend) for e in env}
+        kwargs['additional_env'] = env
+    args = getattr(entity, 'args', None)
+    if args is not None:
+        args = parse_substitution(args, entity.frontend)
+        new_args = []
+        for arg in args:
+            if isinstance(arg, str):
+                new_args.extend(arg.split(' '))
+            else:
+                new_args.append(arg)
+        args = new_args
+    else:
+        args = []
+    cmd_list = [cmd]
+    cmd_list.extend(args)
```

Comment:
@ivanpauno I think that we're not handling `cmd` properly here, and that this specific case may need a wrapping substitution that splits up the wrapped substitution evaluation.

Reply:
I think it's enough. Check this example:

```xml
<launch>
    <executable cmd="ls" cwd="/" name="my_ls" args="-l -a -s" args-sep=" " shell="false" output='log'>
        <env name="var" value="1"/>
    </executable>
</launch>
```

I will modify `executable.xml` to look like this.